kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -108,9 +111,13 @@ private void uploadPart(int size) throws IOException {
       log.debug("New multi-part upload for bucket '{}' key '{}'", bucket, key);
       multiPartUpload = newMultipartUpload();
     }
-
+    final int partSize = size;
     try {
-      multiPartUpload.uploadPart(new ByteArrayInputStream(buffer.array()), size);
+      RetryWithExponentialBackoff.blocking(new Runnable() {
```

Comment:
Yeah, I agree that we can skip the class and have a static method in `S3OutputStream`. In the first run I was thinking to add more to the class (eg. different retry strategies but then switch to some other work related stuff :) )